Documentation/Releases.md

```diff
@@ -0,0 +1,113 @@
+   ```
+   This will yield something like `v1.2.3`, where `1.2.3` is the version number
+   and the `v` prefix is to indicate that this is really a version tag.
+4. Compare this with the
+   [latest release](https://github.com/trixi-framework/HOHQMesh/releases/latest)
+   on GitHub. If the version numbers of the latest tag and the latest release
+   differ, investigate (and possibly fix it) before proceeding.
+5. Determine which kind of version increment is necessary based on the changes
+   since the last release. We use [semantic versioning](https://semver.org),
+   which [states](https://semver.org/spec/v2.0.0.html#summary)
+   > Given a version number MAJOR.MINOR.PATCH, increment the:
+   >
+   > MAJOR version when you make incompatible API changes,  
+   > MINOR version when you add functionality in a backwards compatible manner, and  
+   > PATCH version when you make backwards compatible bug fixes.
+
+   To see all the changes since the last release, you can use GitHub's `compare`
+   functionality by navigating to the following address, adjusting the version
+   tag appropriately:  
+   https://github.com/trixi-framework/HOHQMesh/compare/v1.2.3...master
+
+   Let's say that the changes require a *minor* version increment, thus the new
+   release version following `1.2.3` would be `1.3.0`.
+
+## Release creation
+1. Increment the version number in [Source/HOHQMeshMain.f90](../Source/HOHQMeshMain.f90).
+   The current version in the file should be something like `1.2.4-pre`,
+   indicating that the `master` branch is in a pre-release state (as it should be
+   during development). Change the pre-release version string to the new version
+   number you determined above, e.g., to `1.3.0`:
+   ```fortran
+   CHARACTER(LEN=*), PARAMETER :: version           = "1.3.0"
+   ```
+2. Commit and push to `master` (make sure to use *your* version number!):
+   ```bash
+   git add Source/HOHQMeshMain.f90
+   git commit -m 'Increment version to v1.3.0'
+   git push
+   ```
+3. Ensure that you did not accidentally break anything and verify that there is
+   a green checkmark behind the latest commit on
+   https://github.com/trixi-framework/HOHQMesh/commits/master.
+4. Create a new release tarball by executing
+   ```bash
+   ./Utilities/createrelease 1.3.0
+   ```
+   where `1.3.0` again refers to the new release version (*without* the `v`
+   prefix!). This will result in a new file `HOHQMesh-v1.3.0.tar.gz` in the
+   current directory.
+5. Test the new release by executing
+   ```bash
+   ./Utilities/testrelease HOHQMesh-v1.3.0.tar.gz
+   ```
+   If it fails, do *not* just change the files in your current directory!
+   Instead, figure out why the tests fail, fix them, commit and push the changes
+   and start over.
+6. Create a new annotated Git tag and push it to GitHub (make sure to use *your*
+   version number both for the tag and the message!):
+   ```bash
+   git tag -a v1.3.0 -m "HOHQMesh v1.3.0"
+   git push --tags
+   ```
```

Comment:
That's not strictly necessary. As far as I know (and remember from doing this in the past), GitHub will create the version tag automatically when you create a release.

Reply:
Yes. However, this manual process of creating releases has the downside that it can be interrupted by someone else committing in the meantime. However, this way it is ensured that the release tarballs are really created from the tagged version of the sources.

We could also do it differently, eg, creating the release on GH first, then downloading the tagged version and creating the tarballs from it. This seemed to be even more overhead, though.